browser/BUILD.gn

```diff
@@ -284,7 +268,37 @@ source_set("browser_process") {
   }
 
   if (brave_rewards_enabled) {
```

Comment:
can we create a `browser/brave_rewards/BUILD.gn` file in a follow-up?

Reply:
I did do that, but had a lot of problems with circular dependency and talking with @iefremov he suggested to just move it into this main one.